engine/client/drivers/docker.go

```diff
@@ -72,11 +71,12 @@ func (d *dockerDriver) create(ctx context.Context, rec *progrock.VertexRecorder,
 		// auth keychain parses the same docker credentials as used by the buildkit
 		// session attachable.
 		if img, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain), remote.WithUserAgent(opts.UserAgent)); err != nil {
-			fmt.Fprintf(os.Stderr, "failed to resolve image digest: %v\n", err)
+			msg := "falling back to leftover engine"
+			msg += fmt.Sprintf("\nfailed to resolve image digest: %v", err)
 			if strings.Contains(err.Error(), "DENIED") {
-				fmt.Fprintf(os.Stderr, "check your docker ghcr creds, it might be incorrect or expired\n")
+				msg += "\n(check your docker ghcr creds, it might be incorrect or expired)"
```

Comment:
A little surprised that we special-case this since I'm not sure we ever publish private images, but I suppose someone could be pointing to their own registry, which wouldn't necessarily be GHCR at that point.

```suggestion
				progrock.Warn("check your docker registry auth; it might be incorrect or expired")
```